src/averageAges.js

```diff
@@ -55,7 +70,37 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const motherDiff = people.map(person => ({
+    ...person,
+    children: people.filter(one => one.mother === person.name),
+  })).map(person => ({
+    ...person,
+    diffAge: person.children.map(child => child.born - person.born),
+  }));
+
+  const motherOfSon = motherDiff.map(person => ({
+    ...person,
+    children: person.children.filter(item => item.sex === 'm'),
+  })).map(person => ({
+    diffAge: person.children.map(child => child.born - person.born),
+  }));
+
+  const differenceAge = !onlyWithSon
+    ? (motherDiff.map(person => person.diffAge)).flat()
+    : (motherOfSon.map(person => person.diffAge)).flat();
```

Comment:
you need to check, not to assign, use `!==`. `flat` is redundant, you return number, not array from `map`.  don't duplicate code